### ISSUE TITLE
Dramatic mAP drop on resume  

### ISSUE THREAD

**Issue by yurkovak** [meituan/YOLOv6#533]
### Before Asking

- [X] I have read the [README](https://github.com/meituan/YOLOv6/blob/main/README.md) carefully. 我已经仔细阅读了README上的操作指引。

- [ ] I want to train my custom dataset, and I have read the [tutorials for training your custom data](https://github.com/meituan/YOLOv6/blob/main/docs/Train_custom_data.md) carefully and organize my dataset correctly; (FYI: We recommand you to apply the config files of xx_finetune.py.) 我想训练自定义数据集，我已经仔细阅读了训练自定义数据的教程，以及按照正确的目录结构存放数据集。（FYI: 我们推荐使用xx_finetune.py等配置文件训练自定义数据集。）  

- [X] I have pulled the latest code of main branch to run again and the problem still existed. 我已经拉取了主分支上最新的代码，重新运行之后，问题仍不能解决。

### Search before asking

- [X] I have searched the YOLOv6 [issues](https://github.com/meituan/YOLOv6/issues) and found no similar questions.

### Question

Hi! I had to interrupt training mid-run (30th epoch) and now I'm trying to resume it and get a much lower mAP in just 1 epoch. 

### Additional

Using latest main. The mAP prior to interruption was 
``` 
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.327
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.473
...
```
and after resume in just one epoch it drops and doesn't restore.
```
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.299
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.438
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.324
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.155
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.337
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.406
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.282
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.491
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.538
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.302
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.609
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.720
Epoch: 31 | mAP@0.5: 0.43843574442675665 | mAP@0.50:0.95: 0.2994665819333631
```
I did some debugging and found out that:
- It is different than the mAP I get if I don't interrupt training;
- If on resume I run validation prior to any training, I get a correct mAP, it drops already after one epoch;
- If I skip optimizer and EMA update mAP stays the same (because EMA state is undamaged);
- I also tried saving and reloading GradScaler's state dict, as expected, it doesn't solve the problem; 
- All in all, it looks like something with the saved model or optimizer state.

The commands I'm using
`python3 -m torch.distributed.launch --master_port=1234 --nproc_per_node 8 tools/train.py --batch 256 --conf-file configs/yolov6s.py --data data/coco.yaml --device 0,1,2,3,4,5,6,7 --eval-interval 10`
Then I change eval-interval to 1 in args.yaml and run
`python3 -m torch.distributed.launch --master_port=1234 --nproc_per_node 8 tools/train.py --batch 256 --conf-file configs/yolov6s.py --data data/coco.yaml --device 0,1,2,3,4,5,6,7 --resume /path/to/YOLOv6/runs/train/v6-s-debug-resume/weights/last_ckpt.pt`

This is how the curve of a resumed run looks *
![Screen Shot 2022-10-03 at 14 08 28](https://user-images.githubusercontent.com/18491807/193562938-87fcf0a6-ae8e-4e41-87ec-ae3ad06d18b2.png)
*tested in a separate branch, verified it's present in the latest main

**Comment by mtjhl** [comment]
Did you train the model on coco dataset ? I will check the  codes about resume and try to reproduce the problem.

**Comment by yurkovak** [comment]
Yes, on COCO, the commands I left use unchanged code and configs from main.
Thanks a lot, looking forward!

**Comment by yurkovak** [comment]
Hi again! Was wondering if you were able to reproduce the issue and if yes then if there are any news/ideas/debugging directions. I wasn't able to localize the problem, so any thoughts would be helpful

**Comment by Heath-zyl** [comment]
same problem.any thoughts would be helpful

**Comment by abdulghani91** [comment]
I have the same problem too.

**Comment by drfarasat** [comment]
I have the same problem. Training yolov6n on single class on custom dataset.  On resuming the map drops significantly. Any help or ideas?
